src/win/getaddrinfo.c

```diff
@@ -380,3 +382,46 @@ int uv_getaddrinfo(uv_loop_t* loop,
   }
   return uv_translate_sys_error(err);
 }
+
+int uv_if_indextoname(unsigned int ifindex, char* ifname, size_t* sizep) {
+  NET_LUID interface_luid;
+  char interface_name[NDIS_IF_MAX_STRING_SIZE + 1];
+  int required_len;
+  int ifname_len;
+  int r;
+
+  ifname_len = *sizep;
+
+  r = ConvertInterfaceIndexToLuid(ifindex, &interface_luid);
+  if (r == ERROR_FILE_NOT_FOUND)
+    return UV_ENXIO;
+  else if (r != 0)
+    return UV_EINVAL;
+
+  r = ConvertInterfaceLuidToNameA(&interface_luid,
```

Comment:
Is there a reason to use `ConvertInterfaceLuidToNameA()` instead of `ConvertInterfaceLuidToNameW()`? My experience has been that on Windows, libuv uses the `W` variants of these functions, and then converts them.